keploy/keploy.go

```diff
@@ -493,32 +542,88 @@ func (k *Keploy) check(runId string, tc models.TestCase) bool {
 	}
 
 	// test application reponse
-	r, err := http.NewRequest("POST", k.cfg.Server.URL+"/regression/test", bytes.NewBuffer(bin))
-	if err != nil {
-		k.Log.Error("failed to create test request request server", zap.String("id", tc.ID), zap.String("url", tc.URI), zap.Error(err))
-		return false
-	}
-	k.setKey(r)
-	r.Header.Set("Content-Type", "application/json")
+	if k.cfg.Server.GrpcEnabled {
+		var r *proto.TestResponse
+		var err error
+		switch tc.Type{
+		case string(models.HTTP):
+			r, err = k.grpcClient.Test(k.Ctx, &proto.TestReq{
+				ID:    tc.ID,
+				AppID: k.cfg.App.Name,
+				RunID: runId,
+				Resp: &proto.HttpResp{
+					Body:          resp.Body,
+					Header:        utils.GetProtoMap(resp.Header),
+					StatusCode:    int64(resp.StatusCode),
+					StatusMessage: resp.StatusMessage,
+					ProtoMajor:    int64(resp.ProtoMajor),
+					ProtoMinor:    int64(resp.ProtoMinor),
+					Binary:        resp.Binary,
+				},
+				TestCasePath: k.cfg.App.TestPath,
+				MockPath:     k.cfg.App.MockPath,
+			})
+			if err != nil {
+				k.Log.Error("failed to create test request request server", zap.String("id", tc.ID), zap.String("url", tc.URI), zap.Error(err))
+				return false
+			}
+			case string(models.GRPC_EXPORT):
+				r, err = k.grpcClient.Test(k.Ctx, &proto.TestReq{
+					ID:    tc.ID,
+					AppID: k.cfg.App.Name,
+					RunID: runId,
+					Resp: &proto.HttpResp{
```

Comment:
Please Resp field because this field stores response for http but in grpc, it should be empty.